static/js/components/CandidateList.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import * as candidatesActions from '../ducks/candidates';
+import sourceStyles from "./Source.css";
+import Plot from './Plot';
+import ThumbnailList from './ThumbnailList';
+import CandidateCommentList from './CandidateCommentList';
+import SaveCandidateGroupSelectDialog from './SaveCandidateGroupSelectDialog';
+import FilterCandidateList from './FilterCandidateList';
+
+
+const CandidateList = () => {
+  const { candidates } = useSelector((state) => state.candidates);
+
+  const userGroups = useSelector((state) => state.groups.user);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
```

Comment:
Ari, out of curiosity---is this pattern safe for loading?  What happens if the fetch fails?

Reply:
Curious what you mean by safe here - if it fails, it won't try again until the page is refreshed, if that's what you're referring to

Reply:
(because `candidates` doesn't change)

Reply:
@stefanv I just tested this by making the handler error, and the back-end error is shown as a front-end notification once (as happens automatically with all of our fetch failures), but it doesn't attempt to fetch again until the user reloads the page or clicks "Submit" on the filtering form. That all seems like what should happen to me.